src/coreclr/tools/aot/ILCompiler.LLVM/Compiler/DependencyAnalysis/LLVMCodegenNodeFactory.cs

```diff
@@ -49,17 +49,19 @@ public override IMethodNode RuntimeExportManagedEntrypoint(string name)
 
         internal ExternMethodAccessorNode ExternSymbolWithAccessor(string name, MethodDesc method, ReadOnlySpan<TargetAbiType> sig)
         {
-            Dictionary<string, ExternMethodAccessorNode> map = _externSymbolsWithAccessors;
+            Dictionary<ExternSymbolKey, ExternMethodAccessorNode> map = _externSymbolsWithAccessors;
+            PInvokeMetadata pInvokeMethodMetadata = method.GetPInvokeMethodMetadata();
+            ExternSymbolKey key = new(pInvokeMethodMetadata.Module, name, pInvokeMethodMetadata.Flags.WasmImportLinkage, sig);
```

Comment:
The layering here no longer quite makes sense. Originally, this only served static symbols, so we would go like so:
```
codegen asks LLVMCodegenCompilation
  for an accessor to a static PInvoke (via 'GetExternalMethodAccessor')
    LLVMCodegenCompilation asks ConfigurablePInvokePolicy
      for a direct symbol name for that static PI
    LLVMCodegenCompilation asks LLVMCodegenNodeFactory
      to produce a node representing the accessor, for that symbol name
```
Somewhat non-obviously, but importantly, `method/sig` here are auxiliary information essentially only used for diagnostic purposes. Now we are using `method` to essentially "skip over" the PI policy, which controls symbol resolution.

I think we can use the PI policy mechanism to our advantage and make this a lot cleaner:
1) Create a new `WasmConfigurablePInvokePolicy : ConfigurablePInvokePolicy` that would do override `GenerateDirectCall` somewhat like so:
```
public override bool GenerateDirectCall(MethodDesc method, out string externName)
{
    if (IsWasmImport(method))
    {
        externName = "__wasm_import_" + moduleName + "_" + externName;
        return true;
    }
}
```
2) Leave all of the accessor code essentially unchanged.
3) In LLVM object writer, ask the policy whether the accessor node represents a WASM import. Could be simple as asking whether the symbol name starts with `__wasm_import_`.